lib/matplotlib/figure.py

```diff
@@ -2829,10 +2836,7 @@ def draw_artist(self, a):
         This method can only be used after an initial draw of the figure,
         because that creates and caches the renderer needed here.
         """
-        if self._cachedRenderer is None:
-            raise AttributeError("draw_artist can only be used after an "
```

Comment:
I suspect that this is raising `AttributeError` to maintain exact API compatibility with older versions of the code where `_cachedRenderer` was not an attribute until it was stashed 😱 .